reader.go

```diff
@@ -203,6 +214,13 @@ func (r *Reader) Offset() int64 {
 	return offset
 }
 
+func (r *Reader) HighWatermark() int64 {
```

Comment:
Instead of exposing the concept of a HighWatermark here, should we replace this with `Lag` which exposes the difference between the offset of the last read message and the high watermark? It seems like this is the concept that the program is gonna be interested in.